templates/templates.go

```diff
@@ -11,97 +11,107 @@ import (
-	return string(data)
+	Name               string   `json:"name"`
+	Description        string   `json:"description"`
+	TemplateFilename   string   `json:"templateFilename"`
+	ContentParts       []string `json:"contentParts"`
+	Subject            string   `json:"subject"`
+	EscapeContentParts []string `json:"escapeContentParts"`
 }
 
 func New(templatesPath string) (models.Templates, error) {
 	templates := models.Templates{}
 
-	if template, err := NewCareteamInviteTemplate(templatesPath); err != nil {
+	if template, err := NewGenericTemplate(templatesPath, models.TemplateNameCareteamInvite); err != nil {
 		return nil, fmt.Errorf("templates: failure to create careteam invite template: %s", err)
 	} else {
 		templates[template.Name()] = template
 	}
 
-	if template, err := NewNoAccountTemplate(templatesPath); err != nil {
+	if template, err := NewGenericTemplate(templatesPath, models.TemplateNameNoAccount); err != nil {
 		return nil, fmt.Errorf("templates: failure to create no account template: %s", err)
 	} else {
 		templates[template.Name()] = template
 	}
 
-	if template, err := NewPasswordResetTemplate(templatesPath); err != nil {
+	if template, err := NewGenericTemplate(templatesPath, models.TemplateNamePasswordReset); err != nil {
 		return nil, fmt.Errorf("templates: failure to create password reset template: %s", err)
 	} else {
 		templates[template.Name()] = template
 	}
 
-	if template, err := NewSignupTemplate(templatesPath); err != nil {
+	if template, err := NewGenericTemplate(templatesPath, models.TemplateNameSignup); err != nil {
 		return nil, fmt.Errorf("templates: failure to create signup template: %s", err)
 	} else {
 		templates[template.Name()] = template
 	}
 
-	if template, err := NewSignupClinicTemplate(templatesPath); err != nil {
+	if template, err := NewGenericTemplate(templatesPath, models.TemplateNameSignupClinic); err != nil {
 		return nil, fmt.Errorf("templates: failure to create signup clinic template: %s", err)
 	} else {
 		templates[template.Name()] = template
 	}
 
-	if template, err := NewSignupCustodialTemplate(templatesPath); err != nil {
+	if template, err := NewGenericTemplate(templatesPath, models.TemplateNameSignupCustodial); err != nil {
 		return nil, fmt.Errorf("templates: failure to create signup custodial template: %s", err)
 	} else {
 		templates[template.Name()] = template
 	}
 
-	if template, err := NewSignupCustodialClinicTemplate(templatesPath); err != nil {
+	if template, err := NewGenericTemplate(templatesPath, models.TemplateNameSignupCustodialClinic); err != nil {
 		return nil, fmt.Errorf("templates: failure to create signup custodial clinic template: %s", err)
 	} else {
 		templates[template.Name()] = template
 	}
 
 	return templates, nil
 }
+
+//NewGenericTemplate returns the requested template
+//templateName is the name of the template to be returned
+func NewGenericTemplate(templatesPath string, templateName models.TemplateName) (models.Template, error) {
```

Comment:
Would change the method name to "NewTemplate". This function returns a specific template (according to the template name provided in the args), not a generic template. This is missleading

Reply:
OK